iwyu_test_util.py

```diff
@@ -419,6 +436,112 @@ def _GetLaunchArguments(cc_file):
   return shlex.split(args)
 
 
+def _CheckForCompileTimeMacro(macro):
+    """This assumes that the local clang++ and the include-what-you-use
+    under test are using the same libclang, and therefore the same default libraries.
+
+    Look for the specified macro in the clang output.
+    """
+    txt = '#include <cstddef>'
+
+    result = subprocess.run(['clang', '-dM', '-E', '-x', 'c++', '-'], capture_output=True,
```

Comment:
I believe this should include any IWYU args; e.g. if a test specifies `-stdlib=libc++`, we want that to influence the probe here, right? That would also make a bit more sense if we could use IWYU instead of clang.

Reply:
I just intended this to find the 'default' stdlib of libclang, so I didn't add the IWYU args. I explicitly check the IWYU args separately, which allows me to pick up canonical (or no_mappings). Calling clang, passing IWYU_EXTRA_ARGS means we would have to filter out the `-Iwyu` specific args so we don't confuse it.

If we can get IWYU to output the macros though, then yes, we can just pass the args in and remove the explicit arg check. I'll see whether I can get that working.

I'm thinking I should reduce this commit to handle just the `has_target_support_for`, and we can add more conditions later.